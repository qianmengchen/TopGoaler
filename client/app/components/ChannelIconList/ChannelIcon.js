import React from 'react';
import { Image, TouchableOpacity } from 'react-native';
import * as styles from './styles';

const ChannelIcon = props => {
  const { item } = props;

  return (
    <TouchableOpacity style={styles.list.iconContainer}>
      <Image style={styles.list.icon} source={{ uri: item.image_url }} />
    </TouchableOpacity>
  );
};

export default ChannelIcon;
