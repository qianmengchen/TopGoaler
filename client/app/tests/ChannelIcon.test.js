import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
require('isomorphic-fetch');

import ChannelIcon from '../components/ChannelIconList/ChannelIcon';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing Channel Icon component', () => {
  const item = { icon: null };

  const wrapper = shallow(<ChannelIcon item={item} />, {
    context: { store: mockStore(initialTaskState) }
  });
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should have 1 Image for icons', () => {
    expect(render.find('Image')).toHaveLength(1);
  });
});
