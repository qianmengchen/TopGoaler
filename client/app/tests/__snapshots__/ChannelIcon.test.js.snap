// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Channel Icon component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 40,
          "borderWidth": 1,
          "height": 80,
          "marginHorizontal": 10,
          "width": 80,
        }
      }
    />
  </TouchableOpacity>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessibilityComponentType": undefined,
      "accessibilityLabel": undefined,
      "accessibilityTraits": undefined,
      "accessible": true,
      "children": Array [
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 40,
              "borderWidth": 1,
              "height": 80,
              "marginHorizontal": 10,
              "width": 80,
            }
          }
        />,
        null,
      ],
      "hasTVPreferredFocus": undefined,
      "hitSlop": undefined,
      "isTVSelectable": true,
      "nativeID": undefined,
      "onLayout": undefined,
      "onResponderGrant": [Function],
      "onResponderMove": [Function],
      "onResponderRelease": [Function],
      "onResponderTerminate": [Function],
      "onResponderTerminationRequest": [Function],
      "onStartShouldSetResponder": [Function],
      "style": Array [
        Object {
          "justifyContent": "center",
        },
        Object {
          "opacity": 1,
        },
      ],
      "testID": undefined,
      "tvParallaxProperties": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "uri": undefined,
          },
          "style": Object {
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "marginHorizontal": 10,
            "width": 80,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessibilityComponentType": undefined,
        "accessibilityLabel": undefined,
        "accessibilityTraits": undefined,
        "accessible": true,
        "children": Array [
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "borderRadius": 40,
                "borderWidth": 1,
                "height": 80,
                "marginHorizontal": 10,
                "width": 80,
              }
            }
          />,
          null,
        ],
        "hasTVPreferredFocus": undefined,
        "hitSlop": undefined,
        "isTVSelectable": true,
        "nativeID": undefined,
        "onLayout": undefined,
        "onResponderGrant": [Function],
        "onResponderMove": [Function],
        "onResponderRelease": [Function],
        "onResponderTerminate": [Function],
        "onResponderTerminationRequest": [Function],
        "onStartShouldSetResponder": [Function],
        "style": Array [
          Object {
            "justifyContent": "center",
          },
          Object {
            "opacity": 1,
          },
        ],
        "testID": undefined,
        "tvParallaxProperties": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": undefined,
            },
            "style": Object {
              "borderRadius": 40,
              "borderWidth": 1,
              "height": 80,
              "marginHorizontal": 10,
              "width": 80,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
